datatree/datatree.py

```diff
@@ -991,7 +991,10 @@ def from_dict(
 
         # First create the root node
         root_data = d.pop("/", None)
-        obj = cls(name=name, data=root_data, parent=None, children=None)
+        if not isinstance(root_data, cls):
+            obj = cls(name=name, data=root_data, parent=None, children=None)
+        else:
+            obj = root_data.copy()
```

Comment:
I think there is potential for bugs here - `.copy()` will copy child nodes too, so if `root_data` has children already they will presumably be copied into the new tree. You could also then double-specify the contents of a single node, once as the child of the root and again in the supplied dict.

In fact your previous PR might do this too - I think we should decide what we want the behavior to be: either drop all children of supplied nodes or explicitly test that we keep children too.

Reply:
this is a bit similar to what `xarray` does already when passing `DataArray` objects with coordinates to the `Dataset` constructor / `Dataset.assign*`: those coordinates will be carried over (unintentionally overriding coordinates explicitly passed in the same call).

I'm not sure if we need to copy the behavior, though.

I didn't have a need for it ever since opening this PR, so this is also pretty low-priority for me (as you might have guessed from the long delay)